internal/ethapi/api.go

```diff
@@ -1294,7 +1294,13 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 	if tx.Protected() {
 		signer = types.LatestSignerForChainID(tx.ChainId())
 	} else {
-		signer = types.HomesteadSigner{}
+		// First reply protection (EIP155) was a part of Spurious Dragon.
+		// Need to distinguish the two HFs because Homestead rejects a subset of S values.
+		if config.IsHomestead(big.NewInt(0).SetUint64(blockNumber)) {
+			signer = types.HomesteadSigner{}
+		} else {
+			signer = types.FrontierSigner{}
+		}
```

Comment:
Shouldn't we replace this entire thing (including the clause above) with 
```
signer = types.MakeSigner(config, blockNumber)
```

Reply:
Fixed